### PR TITLE
Adapter config for VS

### PR DESCRIPTION
**Pull request by mbajaman**
### Description
Added separate folder for different tests. You can install the VS Test Adapter here: https://marketplace.visualstudio.com/items?itemName=JohnnyHendriks.ext01
The test discovery can be quirky, but its a QOL bit to be able to see and run tests from IDE. Tests can be run without the Adapter above, they pop up in a separate window when building the solution.

### Trello Ticket
https://trello.com/c/c66tzY9k

### Additional Info
Small QOL update

### CODE REVIEW

**Comment by mbajaman** [comment]
[Add test adaptor config for VS](https://trello.com/c/c66tzY9k/19-add-test-adaptor-config-for-vs)

**Comment by mbajaman** [comment]
Test bed will require further update once I get a test plan going and iron out some testing requirements.

File: Tests/Code/CodeTests.premake5.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    includedirs {
+        "%{prj.location}/../../Vendor/Catch2/single_include/",
+        "%{prj.location}/../../Code/Game/",
+    }
+
+    vpaths {
+        ["Headers"] = { "**.h", "**.hpp" },
+        ["Sources/*"] = { "**.c", "**.cpp" },
+        ["Documentation"] = { "**.md", "**.txt" },
```
Comment (jake-pauls) [inline Tests/Code/CodeTests.premake5.lua:20]: Although I doubt this will matter - I changed the `Engine` and `Game` premake files to 'Docs' to be a bit shorter cause I realized 'Documentation' almost gets cut off in VS 😂

File: Tests/Code/CodeTests.premake5.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
+project "CodeTests"
+    kind "ConsoleApp"
+    dependson { "Catch2", "Engine" }
```
Comment (jake-pauls) [inline Tests/Code/CodeTests.premake5.lua]: I know this was in my `Tests.premake5.lua` file before - but we can remove 'Catch2' from this list now since it's header-only and we don't have to link against it.

File: Tests/Code/CodeTests.premake5.lua (hunk 1 above)
Comment (jake-pauls) [inline Tests/Code/CodeTests.premake5.lua]: Although I doubt this will matter - I changed the `Engine` and `Game` premake files to 'Docs' to be a bit shorter cause I realized 'Documentation' almost gets cut off in VS 😂

File: Tests/Engine/EngineTests.premake5.lua [hunk 3]
```diff
@@ -0,0 +1,25 @@
+project "EngineTests"
+    kind "ConsoleApp"
+    dependson { "Catch2", "Engine" }
```
Comment (jake-pauls) [inline Tests/Engine/EngineTests.premake5.lua]: See above comment for lines 3 and 20.

File: Tests/Engine/ExampleTest2.cpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+		return number1 * number2;
+	}
+}
+
+TEST_CASE("Multiplying numbers", "[multiplication]") {
+	REQUIRE(sad::Multiplication(2, 2) == 4);
+	REQUIRE(sad::Multiplication(3, 3) == 9);
+	REQUIRE(sad::Multiplication(4, 4) == 16);
+	REQUIRE(sad::Multiplication(5, 5) == 24);
+}
```
Comment (jake-pauls) [inline Tests/Engine/ExampleTest2.cpp]: Can we change this to have curly braces on a new line?
```
namespace sad
{
    // Multiplication function here...
}
```

**Review by jake-pauls — CHANGES_REQUESTED** [review]
So, this PR passing CI is due to #6 - turns out my bet scripts wouldn't properly throw errors when they failed! So the scripts threw errors in CI for not finding the tests but it was registered as a success on GitHub. 💯🔥 💯  Smooth on my part, right?

I'm 90% sure the changes that were made in this commit (image below) were correct. However, we just need to change the `shell=True` to `shell=SUBPROCESS_USE_SHELL`. Funnily enough, also my bad. 🪨 🤠 

image

One other thing - can we move the `GameTests.premake5.lua` and `EngineTests.premake5.lua` files into the root `Tests/` folder and update the paths? Just nicely separates our premake scripts from the source code which feels like a good practice.

```
Tests/
| ---- Engine/
| ---- Game/
| ---- EngineTests.premake5.lua
| ---- GameTests.premake5.lua
```

tldr: I suck! but we need to route the python script to find test executables for both `CodeTests` and `EngineTests` in order for CI to pass. Also have some stylistic changes mentioned in comments and regarding the path structure above. 🚀

File: Tests/Code/CodeTests.premake5.lua (hunk 2 above)
Comment (mbajaman) [reply Tests/Code/CodeTests.premake5.lua]: Removed Catch2 as its header-only

File: Tests/Code/CodeTests.premake5.lua (hunk 1 above)
Comment (mbajaman) [reply Tests/Code/CodeTests.premake5.lua]: Change vpaths in premake files to Docs

File: Tests/Engine/ExampleTest2.cpp (hunk 4 above)
Comment (mbajaman) [reply Tests/Engine/ExampleTest2.cpp]: Fixed curly braces (I have to update my VS Studio settings, i changed it to have inline brackets last semester LOL)

**Comment by mbajaman** [comment]
Ope forgot to rename a variable in the py script

**Comment by mbajaman** [comment]
OK 🚀
Let me know if there's anything else that's not consistent with the rest of the project. I'm still learning more about premake :)

**Review by jake-pauls — APPROVED** [review]
pog 🚀 lgtm

**Review by kiwon132445 — APPROVED** [review]

**Review by kiwon132445 — APPROVED** [review]
